beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -339,6 +339,19 @@ func (s *Service) sendBatchRootRequest(ctx context.Context, roots [][32]byte, ra
 	return nil
 }
 
+func (_ *Service) dedupRoots(roots [][32]byte) [][32]byte {
```

Comment:
interesting we don't have anything like this already implemented

Reply:
I killed utils.go and re-homed this here because I refactored out everything else in that file.